### PR TITLE
docs: update dg use cases

### PR DESCRIPTION
**Pull request by ForAeons**
Addresses https://github.com/AY2324S2-CS2103T-F13-1/tp/issues/244

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
+
+    * 3a1. No action is taken.
+
+    * Use case ends.
+
+</box>
+
+<box no-icon type="success" light>
+
+**: Use case: Retrieve a more recent successful command input**
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:594]:
```suggestion
**Use case: Retrieve a more recent successful command input**
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
 
 <box no-icon type="success" light>
 
-**TODO: Use case: Retrieve a previous successful command input**
+**: Use case: Retrieve a previous successful command input**
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:571]:
```suggestion
**Use case: Retrieve a previous successful command input**
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
+2. User presses the retrieve next command hotkey (e.g. down arrow key).
+
+3. The more recent successful command input is shown in the command box.
+   Use case ends.
+
+<span class="semi-bold">Extensions</span>
+
+* 3a. No successful command input that is more recent is available.
+
+    * 3a1. No action is taken.
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:609]: Technically, the text in the command box is blank if no more recent command input is available.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -495,7 +521,13 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:521]: Can we add week number here: `by providing the student's NUSNet ID and week number to mark/unmark the attendance for`.

And then TAPro marks/unmarks the attendance for that student for that week.

We should also add extensions if the student doesn't exist, or if the week doesn't exist, or if the attendance is already marked/unmarked.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -465,7 +485,13 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
 <box no-icon type="success" light>
 
-**TODO: Use case: Find a Student by Name**
+**Use case: Find a Student by Name**
+
+<span class="semi-bold">MSS</span>
+
+1. User requests to find all students by one or more keywords in the student's name.
+
+    Use case ends.
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:494]: This use case is a bit incomplete, we should state that TAPro shows the students that are found.

We can also add an extension that no result is found, so TAPro shows no students.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -495,7 +521,13 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
 <box no-icon type="success" light>
 
-**TODO: Use case: Clear All Data**
+**Use case: Clear All Data**
+
+<span class="semi-bold">MSS</span>
+
+1. User requests to purge all data from TAPro.
+
+   Use case ends.
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:530]: We can add that TAPro successfully clears all data.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -407,15 +407,35 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
 
 <box no-icon type="success" light>
 
-**TODO: Use case: Edit a Student**
+**Use case: Editing a Student**
+
+<span class="semi-bold">MSS</span>
+
+1. User requests to edit an existing student, indicating the student the user wish to edit. The user provides the update details for the student and remaining details are unchanged.
+
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:423]: Perhaps, this use case is a bit incomplete, we can add that TAPro successful edits the student details.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:568]: Perhaps we may want to add that TAPro successfully does something.

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
-**TODO: Use case: Retrieve a previous successful command input**
+**: Use case: Retrieve a previous successful command input**
+
+<span class="semi-bold">MSS</span>
+
+1. User focuses on the command box.
+
+2. User presses the retrieve previous command hotkey (e.g. up arrow key).
+
+3. The previous successful command input is shown in the command box.
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:579]:
```suggestion
3. TAPro displays the successful command input in the command box.
```

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -536,7 +568,47 @@ For all use cases below, the **System** is TAPro and the **Actor** is the user,
...
+
+**: Use case: Retrieve a more recent successful command input**
+
+<span class="semi-bold">MSS</span>
+
+1. User focuses on the command box.
+
+2. User presses the retrieve next command hotkey (e.g. down arrow key).
+
+3. The more recent successful command input is shown in the command box.
```
Comment (tituschewxj) [inline docs/DeveloperGuide.md:602]:
```suggestion
3. TAPro displays a more recent successful command input in the command box.
```

**Review by tituschewxj — CHANGES_REQUESTED** [review]
Thanks for doing this! I have a few nits.

I feel that the use cases are a bit incomplete, as the final step that TAPro successfully does the command is missing.

Also, it may be clearer, if we start line in the use case with the User or TAPro. (only if applicable, as sometimes it may be clearer not to)

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ForAeons) [reply docs/DeveloperGuide.md:521]: Very good point

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (ForAeons) [reply docs/DeveloperGuide.md:650]: I am not sure what you mean titus, do you mind elaborating?

**Review by tituschewxj — APPROVED** [review]
LGTM! Just resolve the remaining nits, and we can merge this.

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (ForAeons) [reply docs/DeveloperGuide.md:652]: I see. I also moved the section on top of addstu use case.

**Pull request merged**